frontend/src/metabase/query_builder/components/view/ConvertQueryModal/utils.ts

```diff
@@ -0,0 +1,20 @@
+import { DatasetQuery } from "metabase-types/api";
+import { checkNotNull } from "metabase/core/utils/types";
+import Question from "metabase-lib/Question";
+
+export function createDatasetQuery(
+  query: string,
+  question: Question,
+): DatasetQuery {
+  const tableId = question.tableId();
+  const collection =
+    typeof tableId === "number"
```

Comment:
I think this won't work for `string` table ids, maybe you could check if it's not undefined/null, e.g. by `tableId != null`?

Reply:
a764f3ad0be1922848a9824489d0f71aede4155c